indexing-service/src/main/java/org/apache/druid/indexing/common/task/CachingLocalSegmentAllocator.java

```diff
@@ -19,62 +19,69 @@
 
 package org.apache.druid.indexing.common.task;
 
-import com.google.common.base.Preconditions;
 import org.apache.druid.data.input.InputRow;
 import org.apache.druid.indexing.common.TaskLock;
 import org.apache.druid.indexing.common.TaskToolbox;
 import org.apache.druid.indexing.common.actions.LockListAction;
 import org.apache.druid.indexing.common.task.IndexTask.ShardSpecs;
 import org.apache.druid.java.util.common.ISE;
-import org.apache.druid.java.util.common.Pair;
 import org.apache.druid.java.util.common.StringUtils;
 import org.apache.druid.segment.realtime.appenderator.SegmentIdWithShardSpec;
 import org.apache.druid.timeline.partition.ShardSpec;
-import org.apache.druid.timeline.partition.ShardSpecFactory;
 import org.joda.time.Interval;
 
-import javax.annotation.Nullable;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.function.Function;
 import java.util.stream.Collectors;
-import java.util.stream.IntStream;
 
 /**
- * Allocates all necessary segments locally at the beginning and reuse them.
+ * Allocates all necessary segments locally at the beginning and reuses them.
+ *
+ * @see DefaultCachingLocalSegmentAllocator
  */
-public class CachingLocalSegmentAllocator implements IndexTaskSegmentAllocator
+class CachingLocalSegmentAllocator implements IndexTaskSegmentAllocator
```

Comment:
What would be the non-default version of `CachingLocalSegmentAllocator`?

Reply:
Preview of `RangePartitionCachingLocalSegmentAllocator`: https://github.com/ccaominh/incubator-druid/blob/backup-superbatch-range-partitioning/indexing-service/src/main/java/org/apache/druid/indexing/common/task/RangePartitionCachingLocalSegmentAllocator.java

"Default" may not be the best name for the class, so I'm open to better suggestions.

Reply:
Thanks. Maybe `HashPartitioningCachingLocalSegmentAllocator`?

Reply:
Renamed